src/client/Microsoft.Identity.Client/Extensions.cs

```diff
@@ -0,0 +1,101 @@
+{
+    /// <summary>
+    /// Extension methods
+    /// </summary>
+    public static class Extensions // TODO MSAL5: move these to their parent types
+    {
+        /// <summary>
+        /// Returns true if MSAL can use a system browser.
+        /// </summary>
+        /// <remarks>
+        /// On Windows, Mac and Linux a system browser can always be used, except in cases where there is no UI, e.g. SSH connection.
+        /// On Android, the browser must support tabs.
+        /// </remarks>
+        public static bool IsSystemWebViewAvailable(this IPublicClientApplication publicClientApplication)
+        {
+            if (publicClientApplication is PublicClientApplication pca)
+            {
+                return pca.IsSystemWebViewAvailable;
+            }
+
+            throw new ArgumentException("This extension method is only available for the PublicClientApplication implementation " +
+                "of the IPublicClientApplication interface.");
+
+        }
+
+        /// <summary>
+        /// Returns true if MSAL can use an embedded webview (browser). 
+        /// </summary>
+        /// <remarks>
+        /// Currently there are no embedded webviews on Mac and Linux. On Windows, app developers or users should install 
+        /// the WebView2 runtime and this property will inform if the runtime is available, see https://aka.ms/msal-net-webview2
+        /// </remarks>
+        public static bool IsEmbeddedWebViewAvailable(this IPublicClientApplication publicClientApplication)
+        {
+            if (publicClientApplication is PublicClientApplication pca)
+            {
+                return pca.IsEmbeddedWebViewAvailable();
+            }
+
+            throw new ArgumentException("This extension method is only available for the PublicClientApplication implementation " +
+                "of the IPublicClientApplication interface.");
+        }
+
+        /// <summary>
+        /// Returns false when the program runs in headless OS, for example when SSH-ed into a Linux machine.
+        /// Browsers (webviews) and brokers cannot be used if there is no UI support. Instead, please use <see cref="PublicClientApplication.AcquireTokenWithDeviceCode(IEnumerable{string}, Func{DeviceCodeResult, Task})"/>
+        /// </summary>
+        public static bool IsDesktopSession(this IPublicClientApplication publicClientApplication)
+        {
+            if (publicClientApplication is PublicClientApplication pca)
+            {
+                return pca.IsDesktopSession();
+            }
+
+            throw new ArgumentException("This extension method is only available for the PublicClientApplication implementation " +
+                "of the IPublicClientApplication interface.");
+        }
+
+        /// <summary>
+        /// Returns true if a broker can be used. 
+        /// </summary>
+        /// <remarks>
+        /// On Windows, the broker (WAM) can be used on Win10 and is always installed.
```

Comment:
we probably want to provide aka.ms links for each of these?
- https://aka.ms/msal-net-wam
- https://aka.ms/msal-brokers
- https://aka.ms/msal-net-ios-broker, https://aka.ms/msal-net-ios-13-broker
- https://aka.ms/Brokered-Authentication-for-Android

Reply:
Done.